pkg/env/env.go

```diff
@@ -80,47 +78,108 @@ func ParseFile(path string) (_ map[string]string, err error) {
-		line := strings.TrimLeft(scanner.Text(), whiteSpaces)
-		// line is not empty, and not starting with '#'
-		if len(line) > 0 && !strings.HasPrefix(line, "#") {
-			if err := parseEnv(env, line); err != nil {
-				return nil, err
-			}
-		}
+	content, err := io.ReadAll(fh)
+	if err != nil {
+		return nil, err
 	}
-	return env, scanner.Err()
+
+	// replace all \r\n and \r with \n
+	text := regexp.MustCompile(`\r\n?`).ReplaceAllString(string(content), "\n")
+	if err := parseEnv(env, text, false); err != nil {
+		return nil, err
+	}
+
+	return env, nil
 }
 
-func parseEnv(env map[string]string, line string) error {
-	data := strings.SplitN(line, "=", 2)
+// parseEnv parse the given content into env format
+func parseEnv(env map[string]string, content string, enforceMatch bool) error {
+	m := envMatch(content)
 
-	// catch invalid variables such as "=" or "=A"
-	if data[0] == "" {
-		return fmt.Errorf("invalid variable: %q", line)
+	if len(m) == 0 && enforceMatch {
+		return fmt.Errorf("invalid variable: %q", content)
 	}
-	// trim the front of a variable, but nothing else
-	name := strings.TrimLeft(data[0], whiteSpaces)
-	if len(data) > 1 {
-		env[name] = data[1]
-	} else {
-		if strings.HasSuffix(name, "*") {
-			name = strings.TrimSuffix(name, "*")
+
+	for _, match := range m {
+		key := match[1]
+		separator := match[2]
+		value := match[3]
+		value = strings.Trim(value, whiteSpaces)
+		value = strings.Trim(value, "\"")
+
+		if strings.HasPrefix(value, "\"") {
+			value = strings.ReplaceAll(value, "\\n", "\n")
+			value = strings.ReplaceAll(value, "\\r", "\r")
+		}
+
+		// KEY*=1 => KEY, *, =1 => KEY*, =, 1
+		if separator == "*" && strings.HasPrefix(value, "=") {
+			key += "*"
+			separator = "="
+			value = strings.TrimPrefix(value, "=")
+		}
+
+		switch separator {
+		case "=":
+			// KEY=
+			if value == "" {
+				if val, ok := os.LookupEnv(key); ok {
+					env[key] = val
+				}
+			} else {
+				env[key] = value
+			}
+		case "*":
 			for _, e := range os.Environ() {
 				part := strings.SplitN(e, "=", 2)
 				if len(part) < 2 {
 					continue
 				}
-				if strings.HasPrefix(part[0], name) {
+				if strings.HasPrefix(part[0], key) {
 					env[part[0]] = part[1]
 				}
 			}
-		} else if val, ok := os.LookupEnv(name); ok {
-			// if only a pass-through variable is given, clean it up.
-			env[name] = val
 		}
 	}
 	return nil
 }
+
+func envMatch(content string) [][]string {
+	// Form: https://github.com/motdotla/dotenv/blob/aa03dcad1002027390dac1e8d96ac236274de354/lib/main.js#L9C76-L9C76
+	// ([\w.-]+) match key
+	// ([\w.-]+)(\s*[=|*]\s*?|:\s+?) match separator
+	// Remaining match value
+	// e.g. KEY=VALUE => KEY, =, VALUE
+	//
+	//	KEY= => KEY, =, ""
+	//	KEY* => KEY, *, ""
+	//	KEY*=1 => KEY, *, =1
+	re := regexp.MustCompile(
+		`(?m)(?:^|^)\s*(?:export\s+)?([\w.-]+)(\s*[=|*]\s*?|:\s+?)(\s*'(?:\\'|[^'])*'|\s*"(?:\\"|[^"])*"|\s*` +
+			"`(?:\\`|[^`])*`" + `|[^#\r\n]+)?\s*(?:#.*)?(?:$|$)`,
+	)
```

Comment:
Compared to the regex used in dotenv, some modifications have been made here. `(?:\s*=\s*?|:\s+?)` has been changed to `(\s*[=|*]\s*?|:\s+?)`. This modification is made because we need to support `*` and keep the separator for subsequent logical judgments.

```diff
-(?:\s*=\s*?|:\s+?)
+(\s*[=|*]\s*?|:\s+?)
```